esmvalcore/preprocessor/_other.py

```diff
@@ -58,6 +59,88 @@ def clip(cube, minimum=None, maximum=None):
     return cube
 
 
+@preserve_float_dtype
+def cumsum(
+    cube: Cube,
+    coord: Coord | str,
+    weights: np.ndarray | da.Array | bool | None = None,
+    method: Literal["sequential", "blelloch"] = "sequential",
+) -> Cube:
+    """Calculate cumulative sum of the elements along a given coordinate.
+
+    Parameters
+    ----------
+    cube:
+        Input cube.
+    coord:
+        Coordinate over which the cumulative sum is calculated. Must be 1D.
+    weights:
+        Weights for the calculation of the cumulative sum. Each element in the
+        data is multiplied by the corresponding weight before summing. Can be
+        an array of the same shape as the input data, ``False`` or ``None`` (no
+        weighting), or ``True``. The latter is only allowed if
+        ``coord="time"``. Here, the lengths of time intervals will be used as
+        weights, which are automatically calculated from the input data.
+    method:
+        Method used to perform the cumulative sum. Only relevant if the cube
+        has `lazy data
+        <https://scitools-iris.readthedocs.io/en/stable/userguide/
+        real_and_lazy_data.html>`__. See :func:`dask.array.cumsum` for details.
+
+    Returns
+    -------
+    Cube
+        Cube of cumulative sum. Has same dimensions and coordinates of the
+        input cube.
+
+    Raises
+    ------
+    ValueError
+        ``weights=True`` and ``coord!="time"``.
+    iris.exceptions.CoordinateMultiDimError
+        ``coord`` is not 1D.
+    iris.exceptions.CoordinateNotFoundError
+        ``coord`` is not found in ``cube``.
+
+    """
+    cube = cube.copy()
+
+    # Only 1D coordinates are supported
+    coord = cube.coord(coord)
+    if coord.ndim > 1:
+        raise CoordinateMultiDimError(coord)
+
+    # Weighting, make sure to adapt cube units in this case
+    if weights is True and coord.name() != "time":
```

Comment:
Why this limitation? Supporting any 1D coordinates with bounds or multidimensional coordinates with corresponding cell areas seems straightforward.

Reply:
Good point! Done in https://github.com/ESMValGroup/ESMValCore/pull/2642/commits/08ca8232c0e7c04d644806da3b6798dab84bb8c4.